src/Flexdis86/Assembler.hs

```diff
@@ -591,10 +591,9 @@ encodeValue v =
           Addr_64 seg Nothing Nothing _ | seg `elem` [FS, GS] -> return 0x04
           Addr_32 seg Nothing Nothing _ | seg `elem` [FS, GS] -> return 0x04
 
-          -- If there is no base and no index at all, 0x5 indicates
-          -- that ModR/M is followed by a raw displacement.
-          Addr_64 _ Nothing Nothing _ -> return 0x5
-          Addr_32 _ Nothing Nothing _ -> return 0x5
+          -- 0x4 indicates that the Mod/RM byte will be followed by a SIB
+          Addr_64 _ Nothing Nothing _ -> return 0x4
+          Addr_32 _ Nothing Nothing _ -> return 0x4
```

Comment:
I think I buy this case.  A value of 0x5 in R/M is used for IP-relative addressing.  That is handled in the next case (see `ripRefComponents`, so I'm not worried about this changing from 0x5 to 0x4 incorrectly.  Since this is a non-IP-relative memory reference, this makes sense (and the matching in `memRefComponents` makes sure that we are in that correct case).

Reply:
Actually, I have an extra question: can we try this fix with the following test case:

`mov qword ptr [rbp + 0x601000], rdi`

That is testing the Mod=0b10 case corresponding to the table entry we were trying to fix